src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -570,12 +523,73 @@ export function QueryPageContainerComponent(): JSX.Element {
                 countedIsolatesResponse.data;
             const nrOfSelectedIsolates = isolateCountProp.totalNumberOfIsolates;
             const isolateCountGroups = isolateCountProp.groups;
-            setDataContext(isolateCountGroups, nrOfSelectedIsolates);
+            setDataContext(isolateCountGroups, nrOfSelectedIsolates, uniqueValues);
             setNrOfSelectedIsol(nrOfSelectedIsolates);
         }
         setDataIsLoading(false);
     };
 
+    const fetchNewFilterValues = async (): Promise<void> => {
+        setConditionalValuesAreLoading(true);
+        const newUniqueValues = _.cloneDeep(uniqueDataValues);
+        const filterStatusList: number[] = [];
+        const newFilterFor: string[] = [...filter.displayedFilters]
+        if (rowAttribute !== "") {
+            newFilterFor.push(rowAttribute)
+        }
+        if (colAttribute !== "") {
+            newFilterFor.push(colAttribute)
+        }
+
+        const getConditionalFilterValues = newFilterFor.map(
+            async (displayedFilter) => {
+                const urlParams = new URLSearchParams(history.location.search);
+                urlParams.delete("row");
+                urlParams.delete("column");
+                urlParams.delete(displayedFilter);
+
+                const selectedFilterString = urlParams.toString();
+
+                const conditionalFilterUrl = `${FILTER_URL}/${displayedFilter}?${selectedFilterString}`;
+
+                const filterResponse: ApiResponse<FilterConfigDTO> = await callApiService(
+                    conditionalFilterUrl
+                );
+
+                filterStatusList.push(filterResponse.status);
+
+                if (filterResponse.data) {
+                    const filterProp: FilterConfigDTO = filterResponse.data;
+
+                    const uniqueValuesObject: FilterInterface = generateUniqueValuesService(
+                        filterProp
+                    );
+                    newUniqueValues[displayedFilter] =
+                        uniqueValuesObject[displayedFilter];
+                }
+            }
+        );
+
+        await Promise.all(getConditionalFilterValues);
+        
+        let finalFilterStatus = 200;
+        filterStatusList.forEach((status) => {
+            if (status !== 200) {
+                finalFilterStatus = status;
+            }
+        });
+        if (finalFilterStatus === 200) {
+            setUniqueDataValues(newUniqueValues);
+            await fetchIsolateCounted(newUniqueValues);

```

Comment:
Reject: This has nothing to do with a fetchNewFilterValues function. You added a side effect to this function that is in transparent to the rest of the code. To options to fix this:
First: Rename the fetchNewFilterValues to fetchNewFilterValuesAndIsolateCounted
Second (perhaps better one): Add a new function fetchNewFilterValuesAndIsolateCounted: 
```
const async fetchNewFilterValuesAndIsolateCounted() => {
  await fetchNewFilterValues()
  if(filterStatus === 200) {
    await fetchIsolateCounted()
  }
}
```

Reply:
Done by refactoring the container.